dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp

```diff
@@ -85,75 +141,187 @@ void CommandBarFlyoutCommandBar::OnApplyTemplate()
+                                        {
+                                            focusedControl = secondaryCommandAsControl;
+                                        }
+                                        else if (focusedControl && IsControlFocusable(secondaryCommandAsControl, false /*checkTabStop*/) &&
+                                                 focusedControl != secondaryCommandAsControl)
+                                        {
+                                            if (FocusControl(
+                                                    secondaryCommandAsControl /*newFocus*/,
+                                                    focusedControl /*oldFocus*/,
+                                                    winrt::FocusState::Keyboard /*focusState*/,
+                                                    true /*updateTabStop*/))
+                                            {
+                                                args.Handled(true);
+                                                return;
+                                            }
+                                        }
+                                    }
+                                }
+                                loopCount++; // Looping again is needed when the last command has focus
+                            }
+                            while (loopCount < 2 && focusedControl && deltaIndex == 1);
+                        }
+                        args.Handled(true);
+                        break;
+                    }
+                    }
+                }
+            });
+        }
+        else
+        {
+            // UIElement.PreviewKeyDown is not available before RS3. Skip the custom Down/Up arrows handling above.
+            secondaryItemsRoot.AddHandler(winrt::UIElement::KeyDownEvent(), m_keyDownHandler, true);
+        }
     }
 
     if (m_openingStoryboard)
     {
         m_openingStoryboardCompletedRevoker =
-            m_openingStoryboard.get().Completed(winrt::auto_revoke, {
+            m_openingStoryboard.get().Completed(winrt::auto_revoke,
+            {
                 [this](auto const&, auto const&) { m_openingStoryboard.get().Stop(); }
             });
     }
 
     if (m_closingStoryboard)
     {
         m_closingStoryboardCompletedRevoker =
-            m_closingStoryboard.get().Completed(winrt::auto_revoke, {
+            m_closingStoryboard.get().Completed(winrt::auto_revoke,
+            {
                 [this](auto const&, auto const&) { m_closingStoryboard.get().Stop(); }
             });
     }
 }
 
-void CommandBarFlyoutCommandBar::DetachEventHandlers(bool useSafeGet)
+void CommandBarFlyoutCommandBar::DetachEventHandlers(
+    bool useSafeGet)
 {
+    COMMANDBARFLYOUT_TRACE_INFO(*this, TRACE_MSG_METH, METH_NAME, this);
+
     if (m_keyDownHandler)
     {
-        RemoveHandler(winrt::UIElement::KeyDownEvent(), m_keyDownHandler);
```

Comment:
Why do we not want to remove this handler anymore?

Reply:
The code is now using the protected virtual OnKeyDown method.